parser/src/parser/tests/integrity_constraints.rs

```diff
@@ -361,298 +360,232 @@ fn integrity_constraint_with_indexed_trace_access() {
+    expected.boundary_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![enforce!(eq!(
+            bounded_access!(clk, Boundary::First),
+            int!(0)
+        ))],
+    ));
+    expected.integrity_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![enforce_all!(
+            lc!(((x, expr!(access!(c)))) => eq!(access!(x), add!(access!(a), access!(b))))
+        )],
+    ));
+    ParseTest::new().expect_module_ast(source, expected);
 }
 
 #[test]
 fn ic_comprehension_one_iterable_range() {
     let source = "
+    def test
+
     trace_columns:
         main: [a, b, c[4]]
-    
+
+    public_inputs:
+        inputs: [2]
+
+    boundary_constraints:
+        enf clk.first = 0
+
     integrity_constraints:
         enf x = a + b for x in (1..4)";
 
-    let expected = Source(vec![
-        SourceSection::Trace(vec![vec![
-            TraceBinding::new(Identifier("a".to_string()), 0, 0, 1),
-            TraceBinding::new(Identifier("b".to_string()), 0, 1, 1),
-            TraceBinding::new(Identifier("c".to_string()), 0, 2, 4),
-            TraceBinding::new(Identifier("$main".to_string()), 0, 0, 6),
-        ]]),
-        SourceSection::IntegrityConstraints(vec![Constraint(IntegrityConstraint::new(
-            ConstraintExpr::Inline(InlineConstraintExpr::new(
-                SymbolAccess(SymbolAccess::new(
-                    Identifier("x".to_string()),
-                    AccessType::Default,
-                    0,
-                )),
-                Add(
-                    Box::new(SymbolAccess(SymbolAccess::new(
-                        Identifier("a".to_string()),
-                        AccessType::Default,
-                        0,
-                    ))),
-                    Box::new(SymbolAccess(SymbolAccess::new(
-                        Identifier("b".to_string()),
-                        AccessType::Default,
-                        0,
-                    ))),
-                ),
-            )),
-            Some(vec![(
-                Identifier("x".to_string()),
-                Iterable::Range(Range::new(1, 4)),
-            )]),
-            None,
-        ))]),
-    ]);
-    ParseTest::new().expect_ast(source, expected);
+    let mut expected = Module::new(ModuleType::Root, SourceSpan::UNKNOWN, ident!(test));
+    expected
+        .trace_columns
+        .push(trace_segment!(0, "$main", [(a, 1), (b, 1), (c, 4)]));
+    expected.public_inputs.insert(
+        ident!(inputs),
+        PublicInput::new(SourceSpan::UNKNOWN, ident!(inputs), 2),
+    );
+    expected.boundary_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![enforce!(eq!(
+            bounded_access!(clk, Boundary::First),
+            int!(0)
+        ))],
+    ));
+    expected.integrity_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![enforce_all!(
+            lc!(((x, range!(1..4))) => eq!(access!(x), add!(access!(a), access!(b))))
+        )],
```

Comment:
note: this is another constraint that doesn't make sense. `enf 1 = a + b` would be fine, but if we enforce that `a+b` is 4 different things then only one of the constraints will hold, so constraint evaluation will fail.

Reply:
Yep, I know almost all of these errors were there before, but I think they're actually problematic now, so I figured pointing them all out at this point is the best way to identify additional changes that need handling. Since they're all minor changes I probably prefer to adjust them now since they're all identified in this review, but if you have a strong preference to do it in a follow-up PR I think that's ok.